test/views/behaviour.js

```diff
@@ -0,0 +1,193 @@
+    $('a.publishAllAtOnce').click(function() {
+        publishAllSpecs();
+    });
+    $('a.publishAllWithDelay').click(function() {
+        publishAllSpecs(true);
+    });
+    $('td.metadata').click(function() {
+        var spec = findSpec($(this).attr('data-spec-id'));
+        dumpObject(spec);
+    });
+    $('td.status').click(function() {
+        var job = allJobs[$(this).attr('data-spec-id')];
+        var message = allMessages[$(this).attr('data-spec-id')];
+        dumpObject(job);
+        logMessage(message);
+    });
+    refresh();
+
+}
+
+function refresh() {
+
+    $.get(endpoint() + '/status/', function(data) {
+        updateJobs(data);
+        window.setTimeout(refresh, $('input#rate')[0].value * 1000);
+
+        if ($('input#scroll')[0].checked) {
+            log[0].scrollTop = log[0].scrollHeight;
+        }
+
+  });
+
+}
+
+function updateJobs(data) {
+
+    var spec, id, step, completed;
+
+    for (var i in allSpecs) {
+        id = allSpecs[i].id;
+        spec = data.requests[location.href + 'drafts/' + id];
+        if (spec) {
+            allJobs[id] = allJobs[id] || spec.jobs;
+            completed = 0;
+            allMessages[id] = null;
+            for (var j in JOBS) {
+                step = allJobs[id][JOBS[j]];
+                if (step && step.status && 'ok' === step.status.toLowerCase()) {
+                    completed ++;
+                }
+                else {
+                    allMessages[id] = '“' + id + '”: ' + parseInt(100 * completed / JOBS.length) + '% (stuck in ' + JOBS[j] + ').';
+                    break;
+                }
+            }
+            if (!allMessages[id]) {
+                allMessages[id] = '“' + id + '”: done! ' + completed + ' steps completed.';
+            }
+            $('td[data-spec-id="' + id + '"] > span').css('width', (128 * completed / JOBS.length) + 'px');
+        }
+
+    }
+
+}
+
+function publishOneSpec(id) {
+
+    var spec = findSpec(id);
+    var params = {};
+
+    logMessage('Sumitting spec &ldquo;' + id + '&rdquo; for publication&hellip;');
+    params.url = location.href + 'drafts/' + spec.id;
+    params.decision = 'foo';
+    params.token = '34';
+    $.post(endpoint() + '/request', params, handlePublicationResult);
+
+}
+
+function handlePublicationResult(data) {
+
+    logMessage('Response from server:<br />          ' + data + '');
+
+}
+
+function publishAllSpecs(delay) {
+
+    if (delay) {
+        window.alert('Publishing all with random delays not implemented yet.');
+    }
+    else {
+        logMessage('Sumitting all ' + allSpecs.length + ' specs for publication&hellip;');
+            for (var i in allSpecs) {
+            publishOneSpec(allSpecs[i].id);
+        }
+    }
+
+}
+
+function findSpec(id) {
+
+    var result;
+    var i = 0;
+
+    while (!result && i < allSpecs.length) {
+
+        if (id === allSpecs[i].id) {
+            result = allSpecs[i];
+        }
+
+        i++;
+    }
+
+    return result;
+
+}
```

Comment:
I totally :+1: on this if it's supported by the platform.
I wouldn't introduce new complexity or dependencies just to have nicer statements like that one, though&hellip;